lib/Doctrine/DBAL/Driver/PDOQueryImplementation.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Doctrine\DBAL\Driver;
+
+use Doctrine\DBAL\Driver\PDO\Exception;
+use PDOException;
+use PDOStatement;
+
+use function assert;
+use function func_get_args;
+
+use const PHP_VERSION_ID;
+
+if (PHP_VERSION_ID >= 80000) {
+    /**
+     * @internal
+     */
+    trait PDOQueryImplementation
+    {
+        /**
+         * {@inheritdoc}
+         *
+         * @return PDOStatement
+         */
+        public function query(?string $query = null, ?int $fetchMode = null, mixed ...$fetchModeArgs)
```

Comment:
Are the first two arguments nullable? They aren't [documented](https://www.php.net/manual/en/pdo.query.php) as such.

Reply:
Not in PDO. But I need to fulfill `Doctrine\DBAL\Driver\Connection::query()` as well, so I cannot add mandatory parameters.

Reply:
Makes sense.